### PR TITLE
Generate applyconfiguration for server-side apply

### PR DESCRIPTION
**Pull request by mdbooth**
This adds generation of applyconfiguration using code-generator from k/k.

While the configuration isn't currently too complex, it's more complex than it
will be eventually due to the different versions of k/k used for
code-generation and CAPO itself. This will get much simpler when CAPO is bumped
to v0.31.

### CODE REVIEW

**Comment by EmilienM** [comment]
is it ready for review?

**Comment by mdbooth** [comment]
> is it ready for review?

Yes.

File: .gitignore [hunk 1]
```diff
@@ -146,7 +146,7 @@ kubernetes.tar.gz
 # generated files in any directory
 # TODO(thockin): uncomment this when we stop committing the generated files.
 #zz_generated.*
-zz_generated.openapi.go
+#zz_generated.openapi.go
```
Comment (EmilienM) [inline .gitignore:149]: why not removing it? or comment why we don't want that file.

**Review by EmilienM — APPROVED** [review]
I have a minor comment for the gitignore, otherwise LGTM. Looks bit but a lot is generated. From my little knowledge on that topic, I didn't see any red flag.

/approve

**Pull request merged**